lib/node_modules/@stdlib/random/base/mt19937/lib/factory.js

```diff
@@ -0,0 +1,163 @@
+* @throws {RangeError} must provide a positive integer less than the maximum signed 32-bit integer
+* @returns {Function} Mersenne Twister PRNG
+*
+* @example
+* var mt19937 = factory();
+*
+* var v = mt19937();
+* // returns <number>
+*
+* @example
+* // Return a seeded Mersenne Twister PRNG:
+* var mt19937 = factory( 1234 );
+*
+* var v = mt19937();
+* // returns 822569775
+*/
+function factory( seed ) {
+	var state;
+	var mti;
+	var mt = new Array( N );
+	var s;
+
+	if ( arguments.length ) {
+		if ( !isPositiveInteger( seed ) ) {
+			throw new TypeError( 'invalid input argument. Must provide a positive integer. Value: `' + seed + '`.' );
+		}
+		if ( seed > MAX_SEED ) {
+			throw new RangeError( 'invalid input argument. Must provide a positive integer less than the maximum signed 32-bit integer. Value: `' + seed + '`.' );
+		}
+		state = seed | 0; // asm type annotation
+	} else {
+		state = randint32() | 0; // asm type annotation
+	}
+
+	mt[ 0 ] = state >>> 0;
+	for ( mti = 1; mti < N; mti++ ) {
+		s = mt[ mti - 1 ] ^ ( mt[ mti - 1 ] >>> 30 );
+		mt[ mti ] =
+			( ( ( ( (s & 0xffff0000) >>> 16 ) * 1812433253 ) << 16 ) +
+			( (s & 0x0000ffff) * 1812433253) ) + mti;
+		mt[ mti ] >>>= 0;
+	}
+
+	setReadOnly( mt19937, 'NAME', 'mt19937' );
+	setReadOnly( mt19937, 'SEED', state );
+	setReadOnly( mt19937, 'MIN', 1 );
+	setReadOnly( mt19937, 'MAX', UINT32_MAX - 1 );
+	setReadOnly( mt19937, 'normalized', normalized );
+
+	setReadOnly( normalized, 'NAME', mt19937.NAME );
+	setReadOnly( normalized, 'SEED', mt19937.SEED );
+	setReadOnly( normalized, 'MIN', ( mt19937.MIN - 1.0 ) / NORMALIZATION_CONSTANT );
+	setReadOnly( normalized, 'MAX', ( 4294967294.0 ) * ( 1.0 / 4294967295.0 ) );
+
+	return mt19937;
+
+	/**
+	* Generates a pseudorandom integer on the interval \\( [1,2^{31}-1) \\).
+	*
+	* @private
+	* @returns {PositiveInteger} pseudorandom integer
+	*
+	* @example
+	* var v = mt19937();
+	* // returns <number>
+	*/
+	function mt19937() {
+		var mag01 = [ 0x0, MATRIX_A ];
+		var kk;
+		var y;
+
+		if (mti >= N) {
+			for ( kk = 0; kk < N - M; kk++ ) {
+				y = ( mt[kk] & UPPER_MASK ) | ( mt[kk + 1] & LOWER_MASK );
+				mt[kk] = mt[kk + M] ^ ( y >>> 1 ) ^ mag01[y & 0x1];
+			}
+			for ( ; kk < N - 1; kk++ ) {
+				y = ( mt[kk] & UPPER_MASK ) | ( mt[kk + 1] & LOWER_MASK );
+				mt[kk] = mt[kk + (M - N)] ^ (y >>> 1) ^ mag01[y & 0x1];
+			}
+			y = ( mt[N - 1] & UPPER_MASK ) | ( mt[0] & LOWER_MASK );
+			mt[N - 1] = mt[M - 1] ^ ( y >>> 1 ) ^ mag01[y & 0x1];
+			mti = 0;
+		}
+
+		y = mt[ mti ];
+		mti += 1;
+
+		/* Tempering */
+		y ^= y >>> 11;
+		y ^= ( y << 7 ) & 0x9d2c5680;
+		y ^= ( y << 15 ) & 0xefc60000;
+		y ^= y >>> 18;
+
+		return y >>> 0;
+	}
+
+	/**
+	* Generates a pseudorandom number on the interval \\( [0,1) \\).
+	*
+	* @private
+	* @returns {number} pseudorandom number
+	*
+	* @example
+	* var v = normalized();
+	* // returns <number>
+	*/
+	function normalized() {
+		return ( mt19937() - 1 ) * ( 1.0 / 4294967295.0 );
```

Comment:
I would define a normalization constant and elevate to the top level scope. Otherwise, we have to assume that the compiler will be smart enough to not repeatedly evaluate a constant.